### PR TITLE
chore(upgrades): updating .aws packages and es-linter

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

* Update the tf modules to the latest pattern and package
* Update the root linter so dependabot moves again

### CODE REVIEW

File: .aws/cdktf.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "language": "typescript",
-  "app": "npm run --silent compile && node dist/main.js"
-}
+  "app": "npm run --silent compile && node dist/main.js",
+  "projectId": "263a2617-ef8b-4500-a05f-e5dcc79774d9"
```
Comment (bassrock) [inline .aws/cdktf.json:4]: latest cdktf adds this on a synth

File: .aws/src/main.ts [hunk 2]
```diff
@@ -7,10 +7,9 @@ import {
 } from 'cdktf';
 import {
   AwsProvider,
-  DataAwsCallerIdentity,
-  DataAwsKmsAlias,
-  DataAwsRegion,
-  DataAwsSnsTopic,
+  datasources,
```
Comment (bassrock) [inline .aws/src/main.ts:10]: tf modules 3 switches to namespaces

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
       # Run Linter against code base #
       ################################
       - name: Lint Code Base
-        uses: docker://github/super-linter:v3
+        uses: docker://ghcr.io/github/super-linter:slim-v4
```
Comment (bassrock) [inline .github/workflows/linter.yml:56]: superlinter 4 doesnt error out with the latest eslint and typescript.

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM node:17
+FROM node:17-bullseye-slim@sha256:2ce3d9c4d432e462e9f0b4c20f531203221e35e339c98e0a895821dc95c64098
```
Comment (bassrock) [inline Dockerfile:1]: Per SRE swapping to sha pinned docker images. See https://github.com/Pocket/parser-graphql-wrapper/pull/409

File: package.json [hunk 5]
```diff
@@ -28,14 +28,14 @@
...
   },
   "devDependencies": {
-    "@types/jest": "^27.0.3",
-    "@types/node": "^16.11.13",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
-    "@typescript-eslint/parser": "^4.33.0",
-    "eslint": "^7.30.0",
+    "@types/jest": "^27.4.0",
+    "@types/node": "^17.0.8",
+    "@typescript-eslint/eslint-plugin": "^5.9.1",
```
Comment (bassrock) [inline package.json:33]: Updating all the dev dependencies to get past build, linter and test failures.

**Review by jpetto — APPROVED** [review]
ty for this chore work 😄

**Pull request merged**